### PR TITLE
Add feature omdb search

### PR DESCRIPTION
**Pull request by nikos83**
- Add httparty gem and api key to credentials
- Add service to handle API requests
- Add search inside new film form
- Insert search results inside input new film form + show poster
- Add js to catch paste action ( paste image from clipboard to the film poster)
- Delete unused code and correct view.

### CODE REVIEW

File: app/assets/javascripts/pasteAttachment.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+var myOnload =   window.onload = (event) => {
+  if (!(document.getElementById('new_film') === null )) {
+    const form = document.getElementById('new_film');
+    const fileInput = document.getElementById('film_cover_img');
+
+    fileInput.addEventListener('change', () => {
+      console.log('hello');
```
Comment (kamilsdz) [inline app/assets/javascripts/pasteAttachment.js]: remove it

File: app/controllers/films_controller.rb [hunk 2]
```diff
@@ -16,9 +16,16 @@ def show
 
   def new
     @film = Film.new
+    @film_search = OmdbService.new(params[:search], params[:year])
+    return unless @film_search
```
Comment (kamilsdz) [inline app/controllers/films_controller.rb]: I would use null object instead of returning like you did

File: app/services/omdb_service.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class OmdbService
+  include HTTParty
+  base_uri 'www.omdbapi.com'
+
+  def initialize(title, year)
+    @options = { query: { t: title, y: year, apikey: Rails.application.credentials.dig(:omdb, :apikey) } }
```
Comment (kamilsdz) [inline app/services/omdb_service.rb]: I am not sure if this is the right place to build a query

File: app/services/omdb_service.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  end
+
+  def search
+    self.class.get('', @options)
```
Comment (kamilsdz) [inline app/services/omdb_service.rb]: `''` looks weird - how about creating CONST (for example: `ROOT_PATH = '/'`)?

File: app/assets/javascripts/pasteAttachment.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+var myOnload =   window.onload = (event) => {
+  if (!(document.getElementById('new_film') === null )) {
```
Comment (kamilsdz) [inline app/assets/javascripts/pasteAttachment.js]: getting elements by id probably isn't a good idea

File: app/controllers/films_controller.rb [hunk 6]
```diff
@@ -16,9 +16,16 @@ def show
...
+
+    @film.title = @film_search.search.parsed_response['Title']
+    @film.description = @film_search.search.parsed_response['Plot']
   end
 
-  def edit; end
+  def edit
+    @film_search = OmdbService.new(params[:search], params[:year])
```
Comment (kamilsdz) [inline app/controllers/films_controller.rb]: OmdbService called twice - in .new and .edit. How about moving this to a separate method?

Comment (kamilsdz) [reply app/controllers/films_controller.rb]: And also - what if one of these parameters is missing?

File: app/views/films/_form.html.haml [hunk 7]
```diff
@@ -1,3 +1,11 @@
+= form_tag new_film_path, method: :get do
+  %label Search film:
+  .input-group.mb-3
+    = text_field_tag :search, params[:search], placeholder: 'Enter film title', class: 'form-control'
+    = text_field_tag :year, params[:year], placeholder: 'Enter film release year', class: 'form-control'
+    .input-group-append
+      %button.btn.btn-primary search
+%img{:src => "#{ @film_search.search.parsed_response['Poster']}"}/
```
Comment (kamilsdz) [inline app/views/films/_form.html.haml]: You should hide business logic from views - check out decorator pattern

File: config/routes.rb [hunk 8]
```diff
@@ -4,6 +4,7 @@
   resources :films do
     resources :attachments, only: :destroy
     resources :reviews
+    resources :omdb, only: %i[index search]
```
Comment (kamilsdz) [inline config/routes.rb]: do you really need these endpoints?

File: app/services/omdb_service.rb (hunk 4 above)
Comment (kamilsdz) [inline app/services/omdb_service.rb]: add attr_reader for @options

Comment (nikos83) [reply app/services/omdb_service.rb]: base_url is setup above, and in this ' ' you could specific additional urls is this example there's no additional path.

File: app/services/omdb_service.rb (hunk 3 above)
Comment (nikos83) [reply app/services/omdb_service.rb]: I've followed first example form https://github.com/jnunemaker/httparty

File: app/services/omdb_service.rb [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def error
+    result.parsed_response['Error']
+  end
+
+  private
+
+  def inspect_errors(error)
+    puts error.inspect
```
Comment (kamilsdz) [inline app/services/omdb_service.rb]: use `logger` instead of `puts`

File: app/controllers/films_controller.rb [hunk 10]
```diff
@@ -51,6 +54,18 @@ def destroy
 
   private
 
+  def omdb_search
+    return if params.values_at(:search, :year).include?(nil)
+
+    @film_search = FilmDecorator.new(OmdbService.new(params[:search], params[:year]))
```
Comment (kamilsdz) [inline app/controllers/films_controller.rb]: you are wrapping instance of ombd service in decorator. hmm, why like that? I would wrap client response

File: app/services/omdb_service.rb [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+  def initialize(title, year)
+    @options = { query: { t: title, y: year, apikey: Rails.application.credentials.dig(:omdb, :apikey) } }
+  end
+
+  def search
+    self.class.get('', @options)
+  rescue HTTParty::Error => e
+    inspect_errors(e)
+  rescue StandardError => e
+    inspect_errors(e)
```
Comment (kamilsdz) [inline app/services/omdb_service.rb]: How is it handled in controller? Write specs for this.

File: app/controllers/films_controller.rb [hunk 12]
```diff
@@ -51,6 +54,18 @@ def destroy
...
+    insert_data_into_inputs
+  end
+
+  def insert_data_into_inputs
+    @film.title = @film_search.title
+    @film.description = @film_search.plot
```
Comment (kamilsdz) [inline app/controllers/films_controller.rb]: by calling `title` and `plot`, you call API twice!

Comment (kamilsdz) [reply app/controllers/films_controller.rb]: try to refactor whole omdb request in your controller

File: app/controllers/films_controller.rb (hunk 10 above)
Comment (kamilsdz) [reply app/controllers/films_controller.rb]: Also why used you `FilmDecorator` here? I think `Film` is different kind of entity than your API response

File: app/services/omdb_service.rb [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def result
+    search || NullOmdb.new
```
Comment (kamilsdz) [inline app/services/omdb_service.rb]: It won't work when api will return for instance empty string or hash.

File: app/services/omdb_service.rb [hunk 14]
```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+class OmdbService
```
Comment (kamilsdz) [inline app/services/omdb_service.rb]: Add specs for controller, decorator, this service.
And in general I recommend you to read about TDD and how important testing is.

**Review by kamilsdz — CHANGES_REQUESTED** [review]

File: spec/vcr/omdb/nil_params.yml [hunk 15]
```diff
@@ -0,0 +1,59 @@
+---
+http_interactions:
+- request:
+    method: get
+    uri: http://www.omdbapi.com/?apikey=be993178&t=&y=
```
Comment (kamilsdz) [inline spec/vcr/omdb/nil_params.yml]: thx, now I know your API key

**Pull request closed without merging**